exercises/simple-cipher/canonical-data.json

```diff
@@ -0,0 +1,145 @@
+{
+  "exercise": "simple-cipher",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Random key cipher",
+      "cases": [
+        {
+          "description": "Key is made only of lowercase letters",
+          "property": "key",
+          "input": {},
+          "expected": {
+            "match": "^[a-z]+$"
+          }
+        },
+        {
+          "description": "It can encode",
+          "property": "encode",
+          "input": {
+            "plaintext": "aaaaaaaaaa"
+          },
+          "expected": "cipher.key"
```

Comment:
this is interesting because we are to understand that this is not the literal string `"cipher.key"` but instead the key of the cipher, whatever that may be.

This therefore falls under one of the categories of https://github.com/exercism/problem-specifications/issues/1225 . 

This suggests some possible courses of action:

1. Make no change to this PR. It's up to readers of this JSON file to understand that certain fields are symbolic instead of literal strings:
    * The `expected` of every `encode`
    * The `input.ciphertext` of every `decode`
2. Make no change to this PR. It's up to readers of this JSON file to understand that all strings beginning with the seven characters `cipher.` are symbolic instead of literal strings.
3. Make a change to this PR that uses a different type (perhaps some JSON object) for such symbolic inputs and/or outputs.
4. Add your suggestion here

I will not yet express an preference ordering between these options

Reply:
> 3. Make a change to this PR that uses a different type (perhaps some JSON object) for such symbolic inputs and/or outputs.
> 4. Add your suggestion here

As a sort of combination of the above, what about the following:
```json
"expected": {
  "match": "^{cipher.key}$"
}
```

Where `{symbol}` is to be interpolated with local variables before matching is applied.

On the surface, this approach seems to simply add complication to the current syntax. However, if this is to act as a precedent for future situations, the interpolation approach allows multiple variable values to be used. 

Examples of such use:
- `"^{name.first} {name.last}$"`
- `"^{a + b / c}$"`

However, if my fellow maintainers think this is too complicated, my vote would be option 1.

Reply:
This one is very specific to the canonical-data specification/syntax. I believe you guys know way more than me about where the we are and where we should go. I'm completely open to suggestions.